### PR TITLE
added Book and Category tests

### PR DESCRIPTION
**Pull request by DmitriyK86**
coverage

### CODE REVIEW

File: src/test/java/bookstore/controller/BookControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+    @BeforeEach
+    void beforeEach(
+            @Autowired DataSource dataSource,
+            @Autowired WebApplicationContext applicationContext) throws SQLException {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(SecurityMockMvcConfigurers.springSecurity())
+                .build();
+        teardown(dataSource);
```
Comment (okuzan) [inline src/test/java/bookstore/controller/BookControllerTest.java]: teardown in after each should be enough

File: src/test/java/bookstore/controller/CategoryControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.http.MediaType;
+import org.springframework.jdbc.datasource.init.ScriptUtils;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
```
Comment (okuzan) [inline src/test/java/bookstore/controller/CategoryControllerTest.java]: Using EqualsBuilder.reflectionEquals from org.testcontainers.shaded.org.apache.commons.lang3.builder is unusual. The shaded package implies it's from a JAR bundled with Testcontainers, which is not meant for direct use. You should directly depend on Apache Commons Lang for that utility.
import org.apache.commons.lang3.builder.EqualsBuilder;

File: src/test/java/bookstore/service/CategoryServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+        Long categoryId = 1L;
+        Category category = new Category();
+        category.setId(categoryId);
+        category.setName("Fantasy");
+        category.setDescription("Awesome category");
+
+        CategoryDto expected = new CategoryDto();
+        expected.setId(1L);
+        expected.setId(categoryId);
```
Comment (okuzan) [inline src/test/java/bookstore/service/CategoryServiceTest.java]: you set it twice

File: src/test/java/bookstore/controller/CategoryControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+        requestDto.setName("Comedy");
+        String jsonRequest = objectMapper.writeValueAsString(requestDto);
+        MvcResult result = mockMvc.perform(post("/categories")
+                        .content(jsonRequest)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+        CategoryDto expected = new CategoryDto();
+        expected.setName("Comedy");
+        expected.setId(3L);
```
Comment (okuzan) [inline src/test/java/bookstore/controller/CategoryControllerTest.java]: This assumes knowledge about the state of the database that might not be true

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by kozhukhovsky — APPROVED** [review]

**Pull request merged**